### DISCUSSION TITLE
Is the DynamicSelect clause operative in OMEdit?

### DISCUSSION THREAD

**Discussion by mpradovelasco** [OpenModelica/OpenModelica#11157] [Q&A] [answered]
Hi, 

I have tested several examples with the DynamicSelect annotation clause in OMEdit 1.22 - dev without success. I wonder if it is implemented. For example in the following code:
```Modelica fold title:DynamicSelect
annotation (
      Icon(coordinateSystem(preserveAspectRatio=true, extent={{-100,-100},{100,100}}), 
      graphics={Text(origin = {34, -54}, extent = {{-150, 150}, {150, 110}}, textString = "%name"), 
Text(origin = {-57, 36}, extent = {{-7, 12}, {7, -12}}, 
textString = DynamicSelect("J",if (gateType == 3) then "J" elseif gateType== 2 then "D" else "T"))}));
```
I try to change the string assigned to textString according to the variable gateType, which is defined :
``` Modelica fold title:gateType
type FlipFlop = Integer(min=1, max=3) "Types of flip-flop";
    parameter FlipFlop gateType = 3 "Type of flip-flop"
      annotation(choices(choice = 1 "D flip-flop",
                         choice = 2 "T flip-flop",
                         choice = 3 "JK flip-flop"));
```
but the icon only shows the literal "J". 
In the case that it is not implemented, I wonder when it is plan to do it.

Tests performed in OMEdit 1.22-dev (26 august), with Enable Instance API ON and Windows 11.

Thank again for your effort!

**Comment by casella** [comment, marked as answer]
DynamicSelect should be available and [several issues](https://github.com/OpenModelica/OpenModelica/issues?q=is%3Aissue+dynamicselect+is%3Aclosed) regarding it were fixed in the recent past, but maybe there are still bugs in it. Please open a proper ticket about it. Thanks!

**Reply by mpradovelasco** [reply]
OK. I will open a ticket of this issue. Thank you!
